roles/openshift_hosted/tasks/registry/storage/s3.yml

```diff
@@ -0,0 +1,105 @@
+---
+- fail:
+    msg: >
+      openshift_hosted_registry_storage_s3_accesskey and
+      openshift_hosted_registry_storage_s3_secretkey are required
+  when: openshift.hosted.registry.storage.s3.accesskey | default(none) is none or openshift.hosted.registry.storage.s3.secretkey | default(none) is none
+
+- fail:
+    msg: >
+      openshift_hosted_registry_storage_s3_bucket and
+      openshift_hosted_registry_storage_s3_region are required
+  when: openshift.hosted.registry.storage.s3.bucket | default(none) is none or openshift.hosted.registry.storage.s3.region | default(none) is none
+
+# Currently this is relying on the boto config (or env vars) on the master host, it
+# should be a local_action and use separate credentials for creating the
+# bucket (vs the ones for accessing the bucket)
+- name: Create the S3 bucket
+  become: no
+  local_action:
+    module: s3
+    bucket: "{{ openshift.hosted.registry.storage.s3.bucket }}"
+    region: "{{ openshift.hosted.registry.storage.s3.region }}"
+    mode: create
+  when: openshift.hosted.registry.storage.s3.create | default(false) | bool
+
+- name: Test if docker registry config secret exists
+  command: >
+    {{ openshift.common.client_binary }}
+    --config={{ openshift_hosted_kubeconfig }}
+    --namespace={{ openshift.hosted.registry.namespace | default('default') }}
+    get secrets {{ registry_config_secret_name }} -o json
+  register: secrets
+  changed_when: false
+  failed_when: false
+
+- set_fact:
+    registry_config: "{{ lookup('template', '../templates/registry_config.j2') | b64encode }}"
+
+- set_fact:
+    registry_config_secret: "{{ lookup('template', '../templates/registry_config_secret.j2') | from_yaml }}"
+
+# TODO: update config secret if different
+#       need to make sure update will not update backend storage. since that
+#       is destructive
```

Comment:
I'm wondering if it is worth adding a filter here to address this comment, I worry that there will be people that want to go back to add s3 storage after the fact, or want to update things like the chunk size and the behavior being counterintuitive.